# Week 8 :Case Study 4: Faheem Mohammed

In [1]:
# Import all the required Libraries
import pandas as pd
import numpy as np
import scipy.stats as stats

In [2]:
#Load the Data Provided
data= pd.read_csv('Sale add.csv')

In [18]:
#Displaying dataset into the notebook
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [3]:
data

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [19]:
#Describing the basic values of the data
data.describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

# 1.The company wishes to clarify whether there is any increase in sales after stepping into digital marketing

H0:There is no significant difference in sales even after entering into digital marketing.
 

H1:There is significant difference in sales even after entering into digital marketing.

The Level of significance alpha is 0.05

In [29]:
stats.ttest_rel(Sales_After_digital_add(in $),Sales_before_digital_add(in $),alternative='greater')

Ttest_relResult(statistic=12.09070525287017, pvalue=3.168333502287889e-11)

In [30]:
ttest,p_value =stats.ttest_ind(a,b,alternative='greater')

The calculated p value  is less  than tabulated value of 0.05  .Means P_calculated>P_hypothesis.So we reject the null hypothesis and accept the alternative hypothesis

Hence we conclude that there is a significant increase in the sales after stepping into digital marketing.


# 2.The company needs to check whether there is any dependency between the features “Region” and “Manager”.

HO: There is no dependency with the region and the manager 

H1:There is dependency with the region and the manager

Level of significance alpha be 0.05


In [12]:
#Creating the contigency and crosstabs for region and manager from the given data

In [13]:
crosstab = pd.crosstab(data.Manager, data.Region)

In [14]:
crosstab

Region       Region - A  Region - B  Region - C
Manager                                        
Manager - A           4           4           1
Manager - B           3           1           3
Manager - C           3           2           1

In [16]:
#Calculating the statistics and the p value

In [15]:
stats.chi2_contingency(crosstab)

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 2.86363636, 2.04545455],
        [3.18181818, 2.22727273, 1.59090909],
        [2.72727273, 1.90909091, 1.36363636]]))

Conclusion

Calculated p Value is 0.549 which is greater than p=0.05 hence we cannot reject the null hypothesis at 0.05 level of significance at 4 degrees of freedom.So we conclude that there is no 